bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/scheduler/EventForegroundTask.kt

```diff
@@ -0,0 +1,50 @@
+package io.bucketeer.sdk.android.internal.scheduler
+
+import io.bucketeer.sdk.android.internal.di.Component
+import io.bucketeer.sdk.android.internal.event.EventInteractor
+import io.bucketeer.sdk.android.internal.event.SendEventsResult
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+internal class EventForegroundTask(
+  private val component: Component,
+  private val executor: ScheduledExecutorService,
+) : ScheduledTask {
+
+  override var isStarted: Boolean = false
+    private set
+
+  private var scheduledFuture: ScheduledFuture<*>? = null
+
+  private val eventUpdateListener = EventInteractor.EventUpdateListener { _ ->
+    executor.execute {
+      val result = component.eventInteractor.sendEvents(force = false)
+      if (result is SendEventsResult.Success && result.sent) {
+        reschedule()
+      }
+    }
+  }
```

Comment:
`EventForegroundTask` listens to updates from EventInteractor and send events if current  queue exceeds eventsMaxBatchQueueCount.

Reply:
Sure, it should be not that difficult, but I need more information.

- `pollingInterval` is configurable, so users can set any interval short enough(such as 1 minute)
  - what should we do in this case?
- Should we do same error handling for `EventForegroundTask`?

Reply:
implemented retry logic in [`390bd00` (#71)](https://github.com/ca-dp/bucketeer-android-sdk/pull/71/commits/390bd00f7a6327ab3017ff98bf0e17a89913b0d1)

Reply:
Thank you!